### PR TITLE
fix(system-backup): should handle empty backup target

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/5055

### CODE REVIEW

**Review by derekbit — APPROVED** [review]
LGTM

BTW, the validating webhook already checks the inputs, so the UI doesn't any change. right?

**Comment by c3y1huang** [comment]
> BTW, the validating webhook already checks the inputs, so the UI doesn't any change. right?

We still want an error message similar to the Backup page.
Ref: https://github.com/longhorn/longhorn/issues/5055#issuecomment-1350182125

/cc @smallteeths

File: controller/system_backup_controller.go [hunk 1]
```diff
@@ -422,10 +426,7 @@ func (c *SystemBackupController) reconcile(name string, backupTargetClient engin
 		cleanupLocalSystemBackupFiles(tempBackupArchivePath, tempBackupDir, log)
 
 	case longhorn.SystemBackupStateDeleting:
-		err = c.cleanupRemoteSystemBackupFiles(systemBackup, backupTargetClient)
-		if err != nil {
-			return err
-		}
+		cleanupRemoteSystemBackupFiles(systemBackup, backupTargetClient, log)
```
Comment (innobead) [inline controller/system_backup_controller.go:429]: So we will ignore the remote deletion if there is a connection issue with the backup target. It means the SystemBackup CR will be deleted, but later if the connection is backup, the SystemBackup will be resync from the target, and be created with ready state, right?

File: webhook/resources/systembackup/validator.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	if err != nil {
+		return werror.NewBadRequest(err.Error())
+	}
+
+	backupTarget, err := v.ds.GetBackupTargetRO(types.DefaultBackupTargetName)
+	if err != nil {
+		return werror.NewBadRequest(err.Error())
+	}
+
+	backupType, err := util.CheckBackupType(backupTarget.Spec.BackupTargetURL)
```
Comment (innobead) [inline webhook/resources/systembackup/validator.go:52]: This is good, at least we check the backup target setting available or not.

**Review by innobead — APPROVED** [review]

File: controller/system_backup_controller.go (hunk 1 above)
Comment (c3y1huang) [reply controller/system_backup_controller.go:429]: Correct.

**Comment by innobead** [comment]
@c3y1huang Need to backport to 1.4.x.

**Pull request merged**